docs/docs/rules.mdx

```diff
@@ -54,6 +54,28 @@ rules:
 This example rule applies at the start of conversation as well as when the user decides
 to a send a message with an intent `greet` in the middle of an ongoing conversation.
 
+After training, `RulePolicy` will perform a check to make sure that the rules don't
+contradict the stories and each other.
+For example the following two rules are contradicting:
+
+```yaml-rasa
+rules:
+- rule: Chitchat
+  steps:
+  - intent: chitchat
+  - action: utter_chitchat
+
+- rule: Greet instead of chitchat
+  steps:
+  - intent: chitchat
+  - action: utter_greet
+```
+
+:::caution
+Rules are not meant to hardcode a state machine.
+Rules are restricted to have only one user turn, but there can still be multiple
+bot events, including e.g. a form being filled and its subsequent submission.
```

Comment:
i think this whole note might be gibberish to a user (esp. the mention of state machine).

Reply:
I'll let @b-quachtran review anything further

Reply:
I took it from the issue https://github.com/RasaHQ/rasa/issues/6557, and I thought it is rather good. Do you have a better suggestion?

Reply:
i'll let ben help you out with this, but "state machine" is something that users probably don't care about, or know about for that matter

Reply:
oh, I thought it is common notion